### PR TITLE
#59-edit cohort details-Zobeir-Rigi

### PR DESCRIPTION
**Pull request by Zobeir-Rigi**
## Description

Created Update cohort page so the manager can update the cohort details. There was a bug, if there was not any trainee we could not get the milestone so, now we can get the milestone even if there is no trainee. After submitting we can see a message if it was a success or not, put setTimeout so the message will go after 3 seconds. Also query updated ORDER BY cohorts Asc.

## Related to
[Tiket](https://github.com/BoshraM/ldn9-Ctrl-Shift-Learn/issues/59)
Fixes #1

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] I have commented my code
- [x] I have updated any documentation

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -32,8 +33,12 @@ const App = () => {
 				<Route path="/admin/cohorts" element={<Cohorts />} />
 				<Route path="/admin/cohorts/addcohort" element={<CohortForm />} />
 				<Route path="/admin/cohorts/:id" element={<Cohort />} />
+				<Route path="/admin/cohorts/updatecohort/:id" element={<UpdateCohort />} />
```
Comment (PeteLindsell) [inline client/src/App.js:36]: I would go for path="/admin/cohorts/:id/edit/" so you have your entity id and then the action you are doing on it

File: client/src/App.js [hunk 2]
```diff
@@ -32,8 +33,12 @@ const App = () => {
...
 				<Route path="/admin/cohorts/:id/addtrainee" element={<AddTrainee />} />
-				<Route path="/admin/cohorts/:id/updatetrainee/:id" element={<UpdateTrainee />} />
+				<Route
+					path="/admin/cohorts/:id/updatetrainee/:id"
```
Comment (PeteLindsell) [inline client/src/App.js:39]: Same as above for updatetrainee path="/admin/cohorts/:id/trainees/:id/edit" but I see this is not actually part of this PR so dont worry too much about it

File: client/src/component/CohortsTable.js [hunk 3]
```diff
@@ -2,32 +2,39 @@ import { Link } from "react-router-dom";
...
+					<tbody>
+						{cohortsData.map((cohort) => (
+							<tr key={cohort.id}>
+								<td>{cohort.name}</td>
+								<td>
+									<span>
+										<Link to={`${cohort.id}`}>
+											<Button title={"view"} />
+										</Link>
+									</span>
```
Comment (PeteLindsell) [inline client/src/component/CohortsTable.js:24]: It is not valid HTML to put a button in a link as its navigation you want a link here that can be styled as a button, maybe a ButtonLink component. Also is there any need for the span here?

File: client/src/pages/UpdateCohort.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+			const response = await fetch(`/api/cohorts/${id}`, {
+				method: "PUT",
+				headers: {
+					"Content-Type": "application/json",
+				},
+				body: JSON.stringify(input),
+			});
+
+			const data = await response.json();
+			console.log(data);
```
Comment (PeteLindsell) [inline client/src/pages/UpdateCohort.js:34]: please remove console.logs other than possibly for errors which should be console.error()

File: server/controllers/cohortController.js [hunk 5]
```diff
@@ -1,74 +1,121 @@
...
-    } else {
-      throw new Error("Failed to create new cohort.");
-    }
-  } catch (error) {
-    console.error("Error creating cohort:", error);
-    return res.status(500).send("Failed to create new cohort. Please ensure all required fields are provided in the correct format.");
-  }
+	try {
+		const { name, start_date, m_1, m_2, m_3, m_4, m_5, m_6, m_7, m_8 } =
+			req.body; //Destructuring
```
Comment (PeteLindsell) [inline server/controllers/cohortController.js:42]: No need for comments like //Destructuring

**Review by PeteLindsell — COMMENTED** [review]
Looks really good, nice touch on the timed confirmation message but you might want to redirect the user back to the main cohort page after updating, in which case you would want to put the message in context and display in something called a "toast notification" https://www.google.com/search?sxsrf=AB5stBhTi_LI8zbGL8EnsjV6OX5zUO3n6w:1689233517205&q=toast+notification&tbm=isch&sa=X&ved=2ahUKEwjL6PCdlYuAAxUzTaQEHRB5DhIQ0pQJegQIDRAB&biw=1696&bih=920&dpr=2 but probably not one for now

**Review by BoshraM — APPROVED** [review]

**Pull request merged**